app/main.py

```diff
@@ -1,34 +1,214 @@
+class VisibilityError(Exception):
+    def __init__(self) -> None:
+        self.message = "The ship isn't field's area"
+        super().__init__(self.message)
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
+
+    def __repr__(self) -> str:
+        return str(self.__dict__)
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
         # Create decks and save them to a list `self.decks`
-        pass
+        self.decks = self.create_decks(start, end)
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
+    def get_deck(self, row: int, column: int) -> Deck | None:
         # Find the corresponding deck in the list
-        pass
+        decks = self.decks
+        for deck in decks:
+            if deck.row == row and deck.column == column:
+                self.fire(row, column)
+                return deck
 
-    def fire(self, row, column):
+    def fire(self, row: int, column: int) -> None:
         # Change the `is_alive` status of the deck
         # And update the `is_drowned` value if it's needed
-        pass
+        decks = self.decks
+        for deck in decks:
+            if deck.row == row and deck.column == column:
+                deck.is_alive = False
+        decks_values = [deck.is_alive for deck in self.decks]
+        if not any(decks_values):
+            self.is_drowned = True
+
+    @staticmethod
+    def create_decks(start: tuple, end: tuple) -> list[Deck]:
+        result = []
+        if start == end:
+            result.append(Deck(start[0], start[1]))
+            return result
+        if start[0] == end[0]:
+            for i in range(start[1], end[1] + 1):
+                result.append(Deck(start[0], i))
+        if start[1] == end[1]:
+            for i in range(start[0], end[0] + 1):
+                result.append(Deck(i, start[1]))
+        return result
+
+    def __repr__(self) -> str:
+        return str(self.__dict__)
 
 
 class Battleship:
-    def __init__(self, ships):
+    def __init__(self, ships: list[tuple]) -> None:
         # Create a dict `self.field`.
         # Its keys are tuples - the coordinates of the non-empty cells,
         # A value for each cell is a reference to the ship
         # which is located in it
-        pass
+        self.ships = {coords: Ship(coords[0], coords[1]) for coords in ships}
```

Comment:
Why are you using `dict` here? Almost in all other methods you are using `.values()` to get data. So why can't we create simple list?

P.S. I'm not saying that you're wrong, you can prove your point and argue for using `dict`

Reply:
task's condition says to use dictionary